core/coreapi/unixfs.go

```diff
@@ -0,0 +1,65 @@
+package coreapi
+
+import (
+	"context"
+	"io"
+
+	core "github.com/ipfs/go-ipfs/core"
+	coreiface "github.com/ipfs/go-ipfs/core/coreapi/interface"
+	coreunix "github.com/ipfs/go-ipfs/core/coreunix"
+	mdag "github.com/ipfs/go-ipfs/merkledag"
+	uio "github.com/ipfs/go-ipfs/unixfs/io"
+
+	cid "gx/ipfs/QmXfiyr2RWEXpVDdaYnD2HNiBk6UBddsvEP4RPfXb6nGqY/go-cid"
+)
+
+type UnixfsAPI struct {
+	node *core.IpfsNode
+}
+
+func NewUnixfsAPI(n *core.IpfsNode) coreiface.UnixfsAPI {
+	api := &UnixfsAPI{n}
+	return api
+}
+
+func (api *UnixfsAPI) Add(ctx context.Context, r io.Reader) (*cid.Cid, error) {
+	k, err := coreunix.AddWithContext(ctx, api.node, r)
```

Comment:
Agreed, but not in this PR. Happy to convert the unixfs package to using CIDs in another PR.